File: az-function-app/PayOrder/Startup.cs
using System;
using com.opusmagus.bl;
using com.opusmagus.logging;
using Microsoft.Azure.Functions.Extensions.DependencyInjection;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Logging;

[assembly: FunctionsStartup(typeof(com.opusmagus.api.Startup))]
namespace com.opusmagus.api
{
    public class Startup : FunctionsStartup
    {
        public override void Configure(IFunctionsHostBuilder builder)
        {
            builder.Services.AddSingleton<ILogger, OpusMagusLogger>();
            builder.Services.AddSingleton<ILoggerProvider, OpusMagusLoggerProvider>();
            builder.Services.AddSingleton<ProcessOrderCommand, ProcessOrderCommand>();
        }
    }
}

File: java/serverless-api/pay-order/Startup.java
import System;
import com.paypal;
import dk.commentor.bl.command;
import dk.commentor.logger;
import dk.commentor.sl;
import Microsoft.Azure.Functions.Extensions.DependencyInjection;
import Microsoft.Azure.WebJobs;
import Microsoft.Azure.WebJobs.Hosting;
import Microsoft.Extensions.DependencyInjection;
import Microsoft.Extensions.Logging;
import org.openerp;

[assembly:FunctionsStartup(typeof(dk.commentor.starterproject.api.Startup))]
package dk.commentor.starterproject.api
{
    
    public class Startup implements FunctionsStartup
    {
        public override void Configure(IFunctionsHostBuilder builder)
        {
            builder.Services.AddSingleton<Logger, CommentorLogger>();
            builder.Services.AddSingleton<IPaymentService, PayPalService>();
            builder.Services.AddSingleton<IOrderService, OpenERPService>();
            builder.Services.AddSingleton<ProcessOrderCommand>();
            Console.WriteLine("Function started!");
        }
    }
}
